presto-main/src/main/java/com/facebook/presto/sql/gen/PageProcessorCompiler.java

```diff
@@ -99,15 +104,18 @@ public PageProcessorCompiler(Metadata metadata)
     @Override
     public void generateMethods(ClassDefinition classDefinition, CallSiteBinder callSiteBinder, RowExpression filter, List<RowExpression> projections)
     {
+        boolean hasFilter = !filter.equals(TRUE_PREDICATE);
```

Comment:
shouldn't you apply change of same shape to `GenericPageProcessor`.
So the compiled and non-compiled one correspond to each other.